events-api/src/event_traits.rs

```diff
@@ -61,6 +61,23 @@ impl EventSource {
                     destination_replica: destination.to_string(),
                     error,
                 }),
+                ..Default::default()
+            }),
+            ..self
+        }
+    }
+
+    /// Add replica event specific data to event source.
+    pub fn add_replica_data(
```

Comment:
Did you mean to move self?
If so then suggest with_ prefix to make it clearer.

```suggestion
    pub fn with_replica_data(
```

Reply:
Yes I am moving self. Changed it now.